pkg/storkctl/snapshot_test.go

```diff
@@ -95,5 +102,164 @@ func TestGetVolumeSnapshotsOneSnapshot(t *testing.T) {
 snap1     persistentVolumeClaimName   Pending                         Local
 `
 
-	testSnapshotsCommon(t, newGetCommand, cmdArgs, &snapshots, expected)
+	testSnapshotsCommon(t, newGetCommand, cmdArgs, &snapshots, expected, false)
+}
+
+func TestCreateSnapshotsNoSnapshotName(t *testing.T) {
+	cmdArgs := []string{"volumesnapshots"}
+
+	var snapshots snapv1.VolumeSnapshotList
+	expected := "error: Exactly one argument needs to be provided for snapshot name"
+	calledFatal := false
+
+	func() {
+		defer func() {
+			cmdutil.DefaultBehaviorOnFatal()
+		}()
+		cmdutil.BehaviorOnFatal(func(e string, code int) {
+			if calledFatal {
+				return
+			}
+			calledFatal = true
+			require.Equal(t, 1, code, "Unexpected error code")
+			require.Equal(t, expected, e)
+		})
+		testSnapshotsCommon(t, newCreateCommand, cmdArgs, &snapshots, expected, true)
+	}()
+}
+
+func TestCreateSnapshotsNoPVCName(t *testing.T) {
+	cmdArgs := []string{"volumesnapshots", "-p", "", "snap1"}
+
+	var snapshots snapv1.VolumeSnapshotList
+	expected := "error: PVC name needs to be given"
+	calledFatal := false
+
+	func() {
+		defer func() {
+			cmdutil.DefaultBehaviorOnFatal()
+		}()
+		cmdutil.BehaviorOnFatal(func(e string, code int) {
+			if calledFatal {
+				return
+			}
+			calledFatal = true
+			require.Equal(t, 1, code, "Unexpected error code")
+			require.Equal(t, expected, e)
+		})
+		testSnapshotsCommon(t, newCreateCommand, cmdArgs, &snapshots, expected, true)
+	}()
+}
+
+func TestCreateSnapshotsNoNamespace(t *testing.T) {
+	cmdArgs := []string{"volumesnapshots", "-p", "pvc_name", "snap1"}
+
+	var snapshots snapv1.VolumeSnapshotList
+	expected := "error: an empty namespace may not be set during creation"
+	calledFatal := false
+
+	func() {
+		defer func() {
+			// Restore cmdutil behavior.
+			cmdutil.DefaultBehaviorOnFatal()
+		}()
+		cmdutil.BehaviorOnFatal(func(e string, code int) {
+			if calledFatal {
+				return
+			}
+			calledFatal = true
+			require.Equal(t, 1, code, "Unexpected error code")
+			require.Equal(t, expected, e)
+		})
+		testSnapshotsCommon(t, newCreateCommand, cmdArgs, &snapshots, expected, true)
+	}()
+}
+
+func TestCreateSnapshots(t *testing.T) {
+	cmdArgs := []string{"volumesnapshots", "-p", "pvc_name", "snap1"}
+
+	var snapshots snapv1.VolumeSnapshotList
+	expected := `Snapshot snap1 created successfully
+
+`
+	testSnapshotsCommon(t, newCreateCommand, cmdArgs, &snapshots, expected, false)
```

Comment:
Can be done as a separate PR, but the fake API server should store the object and when we do a get we should get the same object back to confirm it was created correctly.